### PR TITLE
Allow testing trust-bundle integration

### PR DESCRIPTION
**Pull request by pierDipi**
This will allow testing cert-manager trust-bundle integration in Eventing https://github.com/knative/eventing/pull/7532

### CODE REVIEW

File: test/e2e-common.sh [hunk 1]
```diff
@@ -19,11 +19,18 @@ set -Eeo pipefail
...
 function test_setup() {
-  kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.11.0/cert-manager.yaml || return $?
-  wait_until_pods_running cert-manager || return $?
+  kubectl apply -f third_party/cert-manager/00-namespace.yaml
+
+  timeout 600 bash -c 'until kubectl apply -f third_party/cert-manager/01-cert-manager.yaml; do sleep 5; done'
+  wait_until_pods_running "$CERT_MANAGER_NAMESPACE" || fail_test "Failed to install cert manager"
+
+  timeout 600 bash -c 'until kubectl apply -f third_party/cert-manager/02-trust-manager.yaml; do sleep 5; done'
+  wait_until_pods_running "$CERT_MANAGER_NAMESPACE" || fail_test "Failed to install cert manager"
```
Comment (creydr) [inline test/e2e-common.sh:33]:
```suggestion
  wait_until_pods_running "$CERT_MANAGER_NAMESPACE" || fail_test "Failed to install trust manager"
```

File: hack/verify-codegen.sh
Comment (creydr) [inline hack/verify-codegen.sh:1]: not really sure, why we need the changes here :thinking:

File: hack/update-cert-manager.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  echo "Updating Cert Manager to version $cert_manager_version"
+  echo "Updating Trust Manager to version $trust_manager_version"
+
+  rm -rf third_party/cert-manager
+  mkdir -p third_party/cert-manager
+
+  helm repo add jetstack https://charts.jetstack.io --force-update
+  kubectl create namespace --dry-run=client cert-manager -oyaml > third_party/cert-manager/00-namespace.yaml
+  helm template -n cert-manager cert-manager jetstack/cert-manager --create-namespace --version "${cert_manager_version}" --set installCRDs=true > third_party/cert-manager/01-cert-manager.yaml
+  helm template -n cert-manager cert-manager jetstack/trust-manager --create-namespace --version "${trust_manager_version}" --set installCRDs=true > third_party/cert-manager/02-trust-manager.yaml
```
Comment (creydr) [inline hack/update-cert-manager.sh:17]: That way we require helm to be installed :/
Any reasons, why we don't use the release bundle via `kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.13.3/cert-manager.yaml`?

Comment (pierDipi) [reply hack/update-cert-manager.sh:17]: Trust-manager doesn't have that option, so using helm for both is more consistent

Comment (pierDipi) [reply hack/update-cert-manager.sh:17]: see https://cert-manager.io/docs/trust/trust-manager/#installation

Comment (creydr) [reply hack/update-cert-manager.sh:17]: Ah ok :/ Makes sense then.

File: hack/verify-codegen.sh
Comment (creydr) [reply hack/verify-codegen.sh:1]: :+1:

**Review by creydr — APPROVED** [review]
/lgtm

**Pull request merged**